### PR TITLE
Remove panic handling testing code for non-panicking methods.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

see small discussion we had here https://github.com/cosmos/ibc-go/pull/4894/files#r1365157212

closes: #XXXX

### Commit Message / Changelog Entry

```text
chore: remove panic checks for non-panicking methods
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Thanks @DimitrisJim 🙏

**Review by charleenfei — APPROVED** [review]
🚀

File: modules/light-clients/08-wasm/types/client_state_test.go [hunk 1]
```diff
@@ -352,30 +334,22 @@ func (suite *TypesTestSuite) TestInitialize() {
...
+				suite.Require().Equal(expClientState, clientStore.Get(host.ClientStateKey()))
 
 				expConsensusState := clienttypes.MustMarshalConsensusState(suite.chainA.Codec, consensusState)
-				suite.Require().Equal(clientStore.Get(host.ConsensusStateKey(clientState.GetLatestHeight())), expConsensusState)
+				suite.Require().Equal(expConsensusState, clientStore.Get(host.ConsensusStateKey(clientState.GetLatestHeight())))
+			} else {
+				suite.Require().ErrorIs(err, tc.expError)
+
+				suite.Require().False(clientStore.Has(host.ClientStateKey()))
+				suite.Require().False(clientStore.Has(host.ConsensusStateKey(clientState.GetLatestHeight())))
```
Comment (colin-axner) [inline modules/light-clients/08-wasm/types/client_state_test.go:352]: as a side note, I think state checks on failed conditions are unnecessary for the same reasons we aren't doing panic checking (baseapp reverts state and that's an assumption we can rely on)

File: modules/light-clients/08-wasm/types/misbehaviour_handle_test.go [hunk 2]
```diff
@@ -184,18 +168,12 @@ func (suite *TypesTestSuite) TestCheckForMisbehaviour() {
 				Data: []byte{1},
 			}
 
+			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.ctx, endpoint.ClientID)
```
Comment (colin-axner) [inline modules/light-clients/08-wasm/types/misbehaviour_handle_test.go:171]: suite.chainA.GetContext()?

**Review by colin-axner — APPROVED** [review]

File: modules/light-clients/08-wasm/types/misbehaviour_handle_test.go (hunk 2 above)
Comment (DimitrisJim) [reply modules/light-clients/08-wasm/types/misbehaviour_handle_test.go:171]: ah, nice! missed this one! we also have https://github.com/cosmos/ibc-go/issues/4949 to catch any ones that are missed.

**Pull request merged**